lib/set.js

```diff
@@ -59,13 +67,12 @@ Set.prototype.addMigration = function(title, up, down){
  */
 
 Set.prototype.save = function(fn){
-  var self = this
-    , json = JSON.stringify(this);
-  fs.writeFile(this.path, json, function(err){
-    if (err) return fn(err);
-
+  var self = this;
+  this.state.save({
```

Comment:
It's strictly adhering to the existing implementation, to keep the change set as small as possible and to make sure that it doesn't break backwards compatibility. :-) It's been a while, but I'm sure that I wouldn't sent off a pull request with failing tests.

If you're not confident with the state persistence in the current implementation, we could add some tests of the current behaviour to prove that this PR is not altering that behaviour.

And actually, it will not break unless you do migrations in a weird way. You should treat migrations as an append only list, which makes your index in that list the only interesting information.